cli.js

```diff
@@ -1,84 +1,132 @@
-      args: commander.args,
-      rawArgs: commander.rawArgs,
-      name: commander.args[commander.args.length - 1]._name,
-    });
-  };
+  if (!options.function) {
+    yargs.showHelp();
+    logger.error('A valid function name is a required argument');
+    if (options._[0] === 'create') {
+      logger.error('The "create" command allows for the --random(r) flag in lieu of a name');
+    }
+    process.exit(1);
+  }
+  await specificHandler(options);
+  process.exit(0);
 };
 
-commander
-  .version(version)
-  .description('Binaris command line interface')
-  .option('-p, --path <path>',
-    // eslint-disable-next-line quotes
-    'Use directory dir. "create" will create this directory if needed.');
-
-commander
-  .command('login')
-  .description('Login to your Binaris account using an API key')
-  .action(actionWrapper(loginHandler));
-
-commander
-  .command('logs <function>')
-  .description('Print the logs of a function')
-  .option('-t, --tail',
-    'Outputs logs in "tail -f" fashion')
-  .action(namedActionHandler(logHandler));
-
-commander
-  .command('create')
-  .description('Create a function from template')
-  .option('-f, --function <name>',
-    'Name of the function to generate. If omitted, a name will be chosen at random')
-  .action(actionWrapper(createHandler));
+yargs
+  .option('path', {
+    alias: 'p',
+    describe: 'Use directory dir. "create" will create this directory if needed.',
+    type: 'string',
+    global: true,
+  })
+  .usage(
+`Binaris command line interface
 
-commander
-  .command('deploy <function>')
-  .description('Deploys a function to the cloud')
-  .action(namedActionHandler(deployHandler));
+Usage: $0 <command> [options]` // eslint-disable-line comma-dangle
+  )
+  .command('create [function] [options]', 'Create a function from template', (yargs0) => {
+    yargs0
+      .usage('Usage: $0 create [function] [options]')
+      .positional('function', {
+        describe: 'Name of the function to generate',
+        type: 'string',
+      })
+      .option('random', {
+        alias: 'r', describe: 'Generate a random name for your function', type: 'boolean',
+      });
+  }, async (argv) => {
+    if (argv.random) {
+      // eslint-disable-next-line no-param-reassign
+      argv.function = makeNameValid(moniker.choose());
+    }
+    await handleCommand(argv, createHandler);
+  })
+  .command('deploy <function> [options]', 'Deploys a function to the cloud', (yargs0) => {
+    yargs0
+      .usage('Usage: $0 deploy <function> [options]')
+      .positional('function', {
+        describe: 'Name of the function to deploy',
+        type: 'string',
+      });
+  }, async (argv) => {
+    await handleCommand(argv, deployHandler);
+  })
+  .command('remove <function> [options]', 'Remove a previously deployed function', (yargs0) => {
+    yargs0
+      .usage('Usage: $0 remove <function> [options]')
+      .positional('function', {
```

Comment:
A shame you have to repeat common arguments' definitions. Should have been possible to reuse the `function` argument's definition.

Reply:
So there actually is some crazy stuff you can do in regards to yargs project setup. You can use inheritance and modules for each specific command definition. That way your commands are more full fledged `~class` definitions. I decided to start with a small scope and use a more simple setup. If we see things getting too redundant and complex we can consider the migration.